cmd/kubeadm/app/cmd/init.go

```diff
@@ -81,13 +81,13 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 		},
 	}
 
-	cmd.PersistentFlags().StringSliceVar(
-		&cfg.API.AdvertiseAddresses, "api-advertise-addresses", cfg.API.AdvertiseAddresses,
-		"The IP addresses to advertise, in case autodetection fails",
+	cmd.PersistentFlags().StringVar(
+		&cfg.API.BindAddress, "apiserver-addresses", cfg.API.BindAddress,
```

Comment:
We should line up the flag names and the names in the config file.  I suggest the flag be "apiserver-bind-address".  And it should be singular as we can only listen on one.